packages/noya-state/src/reducers/layerReducer.ts

```diff
@@ -2,28 +2,129 @@ import Sketch from '@sketch-hq/sketch-file-format-ts';
 import produce from 'immer';
 import { AffineTransform, transformRect } from 'noya-geometry';
 import { uuid } from 'noya-renderer';
+import { IndexPath } from 'tree-visit';
 import * as Layers from '../layers';
 import * as Models from '../models';
 import {
-  getBoundingRect,
+  deleteLayers,
   getCurrentPage,
+  getBoundingRect,
   getCurrentPageIndex,
-  getLayerIndexPathsExcludingDescendants,
+  getSymbolsPageIndex,
   getLayerTransformAtIndexPath,
+  getLayerIndexPathsExcludingDescendants,
 } from '../selectors/selectors';
 import { SelectionType, updateSelection } from '../utils/selection';
 import { ApplicationState } from './applicationReducer';
+import { createPage } from './pageReducer';
 
 export type LayerAction =
   | [type: 'deleteLayer', layerId: string | string[]]
   | [type: 'groupLayer', layerId: string | string[], name: string]
   | [type: 'ungroupLayer', layerId: string | string[]]
+  | [type: 'addSymbol', layerId: string | string[], name: string]
   | [
       type: 'selectLayer',
       layerId: string | string[] | undefined,
       selectionType?: SelectionType,
     ];
 
+const getParentLayer = (pages: Sketch.AnyLayer, lastIndexPath: IndexPath) => {
+  return Layers.access(pages, lastIndexPath.slice(0, -1)) as Sketch.Page;
+};
+const addToParentLayer = (
+  pages: Sketch.AnyLayer,
+  selectedIndexPaths: IndexPath[],
+  element: Exclude<Sketch.AnyLayer, { _class: 'page' }>,
```

Comment:
Howabout:

```ts
const addSiblingLayer = (
  page: Sketch.Page,
  indexPath: IndexPath,
  layer: Layers.ChildLayer,
) => {
  const parent = getParentLayer(page, indexPath);

  parent.layers.splice(indexPath[indexPath.length - 1], 0, layer);
};
```

I changed the param names, and then I think this function is clearer with only a single `IndexPath`. So we can choose the last index path outside of this function

Reply:
Pages can have any except for pages 🤔  'artboard' | 'symbolMaster'

Reply:
Oh yeah, that's true. Ok, your type was better, let's leave it with that